app/javascript/components/record-form/form/subforms/subform-dialog-fields/component.jsx

```diff
@@ -64,8 +65,30 @@ const Component = ({
       }
     }
   }, []);
-
   return fieldsToDisplay.map(subformSectionField => {
+    let calculation_expression = subformSectionField?.calculation?.expression
+    if (calculation_expression) {
+      let count = Object.keys(values).reduce((prev, curr) => {
+        if (!(calculation_expression.sum.indexOf(curr) < 0) && values[curr] !== "") {
+          return prev + 1;
+        }
+
+        return prev;
+      }, 0);
+
+      if (count === 0) {
+        count = 1;
+      }
+
+      const calculatedVal = parseExpression(calculation_expression).evaluate(values);
+
+      const scoreCalField = `${formSection.unique_id.split("subform_section")[0]}score_calc`;
```

Comment:
We shouldn't hard code field names here instead, the field name should come from the field itself.